network/junos/junos_command.py

```diff
@@ -224,29 +138,31 @@ def main():
 
 
     commands = module.params['commands']
+    encoding = module.params['encoding'] or 'text'
+    if module.params['transport'] == 'netconf':
+        encoding = 'xml'
```

Comment:
documentation above claims this is overridden only if not set. Perhaps

    encoding = module.params.get('encoding')
    if encoding is None:
        if module.params['transport'] == 'netconf':
            encoding = 'xml'
        else:
            encoding = 'text'